File: frontend/types/weekly-menu.ts
export type WeeklyMenuMealType = "breakfast" | "lunch" | "dinner";

export type WeeklyMenuRecipe = {
  id: number;
  title: string;
  category: string;
  image_url: string | null;
  default_servings: number;
  cooking_time: number | null;
  difficulty: string;
};

export type WeeklyMenuItem = {
  id: number;
  family_id: number;
  menu_date: string;
  meal_type: WeeklyMenuMealType;
  recipe_id: number;
  servings: number;
  created_at: string;
  updated_at: string;
  recipe: WeeklyMenuRecipe;
};

export type WeeklyMenuDay = {
  menu_date: string;
  items: WeeklyMenuItem[];
};

export type WeeklyMenuWeek = {
  week_start: string;
  week_end: string;
  days: WeeklyMenuDay[];
};

export type WeeklyMenuItemCreatePayload = {
  menu_date: string;
  meal_type: WeeklyMenuMealType;
  recipe_id: number;
  servings: number;
};

export type WeeklyMenuItemServingsPayload = {
  servings: number;
};

export type WeeklyMenuItemResponse = {
  item: WeeklyMenuItem;
};

File: backend/app/schemas/weekly_menu.py
from __future__ import annotations

from datetime import date, datetime
from typing import Literal

from pydantic import BaseModel, ConfigDict, Field

WeeklyMenuMealType = Literal["breakfast", "lunch", "dinner"]


class WeeklyMenuRecipePublic(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: int
    title: str
    category: str
    image_url: str | None
    default_servings: int
    cooking_time: int | None
    difficulty: str


class WeeklyMenuItemPublic(BaseModel):
    model_config = ConfigDict(from_attributes=True)

    id: int
    family_id: int
    menu_date: date
    meal_type: WeeklyMenuMealType
    recipe_id: int
    servings: int
    created_at: datetime
    updated_at: datetime
    recipe: WeeklyMenuRecipePublic


class WeeklyMenuDayPublic(BaseModel):
    menu_date: date
    items: list[WeeklyMenuItemPublic] = Field(default_factory=list)


class WeeklyMenuWeekResponse(BaseModel):
    week_start: date
    week_end: date
    days: list[WeeklyMenuDayPublic] = Field(default_factory=list)


class WeeklyMenuItemCreateRequest(BaseModel):
    menu_date: date
    meal_type: WeeklyMenuMealType
    recipe_id: int = Field(ge=1)
    servings: int = Field(default=2, ge=1, le=20)


class WeeklyMenuItemServingsUpdateRequest(BaseModel):
    servings: int = Field(ge=1, le=20)


class WeeklyMenuItemResponse(BaseModel):
    item: WeeklyMenuItemPublic
